### ISSUE TITLE
get all the examples to work asynchronously using the decorator

### ISSUE THREAD

**Issue by ShootingStarDragon** [AccelQuasarDragon/FastCVApp#25]
From TurboCVApp created by [ShootingStarDragon](https://github.com/ShootingStarDragon): ShootingStarDragon/TurboCVApp#25  

old title: 
How to package to pypi using poetry, already duplicate of #24

**Comment by ShootingStarDragon** [comment]
refer to #34, #14  for the examples

**Comment by ShootingStarDragon** [comment]
~more decorator documentation:~
~https://stackoverflow.com/questions/60907967/how-to-implement-a-data-parallelising-multiprocessing-pool-decorator-in-python~
~https://stackoverflow.com/questions/10370705/running-multiprocessing-inside-decorator~


~https://stackoverflow.com/questions/26576550/using-multiprocessing-with-a-decorated-function-results-in-a-picklingerror~
~https://stackoverflow.com/questions/9336646/python-decorator-with-multiprocessing-fails~
~https://builtin.com/software-engineering-perspectives/python-class-decorator~

moved to: https://github.com/ShootingStarDragon/FastCVApp/issues/35

**Comment by ShootingStarDragon** [comment]
- [x] ~how to window bind in a subprocess?~ (actually don't need it, what u really need is a way to clear the stream on close properly)
- [x] figure out how to cleanly run mediapipe as a subprocess <----- WAY MORE IMPORTANT
- [x] also figured out 2 things: no 1: reading frame isn't as bad as I read it was, no 2: reading from shared memory and cap.isopened is almost 0 time
- [x] how to cancel main process from subprocess (if this is even the right thing to do)
- [x] 1: stop being a while True loop, rather in main process set some data in a shared datatype. When kivy is closed then change that data and it will stop there (DONE!)
- [x] That's it I think
- [x] call the code to clear stuff after the .run() in the build: ~(not important now because stream is opened in main process BUT is helpful to know I don't need to import Window anymore)~ (here is where we set the shared data to be false so main process can read it and clear everything)
- [x] https://stackoverflow.com/a/54501422/16355112
```
If you want your application to simply run things after the GUI has closed, the easiest and smallest approach would be to place any exit code after TestApp().run(). run() creates a endless loop which also clears any event-data from within kivy so it doesn't hang. That endless loop breaks as soon as the window/gui instance dies. So there for, any code after will execute only after the GUI dies too.
```

- [x] rewrite main.py to reflect changes in TEST_kivy_from_subprocess3.py
- [x] have to do this first tbh: https://github.com/ShootingStarDragon/FastCVApp/issues/65
- [ ] now I have to abstract it: example:
- [ ] need to run even when nothing is running

example list

- [ ] background subtraction
- [ ] https://docs.opencv.org/3.4/d1/dc5/tutorial_background_subtraction.html

- [ ] sepia filter
- [ ] https://medium.com/dataseries/designing-image-filters-using-opencv-like-abode-photoshop-express-part-2-4479f99fb35
- [ ] https://amin-ahmadi.com/2016/03/24/sepia-filter-opencv/

- [ ] use this tutorial to properly load haar cascades:  https://stackoverflow.com/questions/70805922/why-does-the-haarcascades-does-not-work-on-opencv
- [ ] face detection with haar cascades
- [ ] https://docs.opencv.org/3.4/db/d28/tutorial_cascade_classifier.html

- [ ] canny edge detection
- [ ] https://docs.opencv.org/4.x/da/d5c/tutorial_canny_detector.html-

- [ ] mediapipe
- [ ] https://www.youtube.com/watch?v=We1uB79Ci-w
- [ ] https://www.youtube.com/watch?v=01sAkU_NvOY
- [ ] another tutorial: https://www.youtube.com/watch?v=riuU9uH7av8
